controller/src/main/java/io/pravega/controller/server/security/auth/handler/impl/AclAuthorizerImpl.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.security.auth.handler.impl;
+
+import io.pravega.auth.AuthHandler;
+import lombok.NonNull;
+
+class AclAuthorizerImpl extends AclAuthorizer {
+
+    @Override
+    public AuthHandler.Permissions authorize(@NonNull AccessControlList accessControlList, @NonNull String resource) {
+        AuthHandler.Permissions result = AuthHandler.Permissions.NONE;
+
+        String resourceDomain;
+        if (resource.indexOf("::") > 0) {
+            resourceDomain = resource.substring(0, resource.indexOf("::"));
+        } else {
+            resourceDomain = "prn"; // default
+        }
+
+        for (AccessControlEntry accessControlEntry : accessControlList.getEntries()) {
+            // Replaces any `*` with `.*`, if it's not already preceded by `.`, for regex processing.
+            // So, `pravega:://*` becomes `pravega:://.*` and `pravega:://scope:*` becomes `pravega:://scope:.*`
+            String aclResourcePattern = accessControlEntry.getResourcePattern().replaceAll("(?!=.)\\*", ".*");
```

Comment:
The code inside the `for` loop does a lot of regex pattern compilation and matching, which is relatively expensive to process. Anyway we can simplify this? Can we precompute something (here or in `AccessControlEntry`)?

Reply:
Sure, that's a very good idea. I've pushed this logic up to when the built-in Auth handler's accounts DB file is loaded during ACE construction time, which happens just once.